city_scrapers/spiders/chi_northwest_home_equity.py

```diff
@@ -0,0 +1,375 @@
+
+        # loop through the meetings minute (past meetings)
+        for post in pastMeetings:
+            meeting = Meeting(
+                title=self._parse_title(post, True),
+                description=self._parse_description(post, True),
+                classification=self._parse_classification(post),
+                start=self._parse_start(post, True, response),
+                end=self._parse_end(post, True, response),
+                all_day=self._parse_all_day(post, True),
+                time_notes=self._parse_time_notes(post, True),
+                location=self._parse_location(post, True, response),
+                links=self._parse_links(post, True),
+                source=self._parse_source(response)
+            )
+
+            meeting["status"] = self._get_status(meeting)
+            meeting["id"] = self._get_id(meeting)
+
+            yield meeting
+
+        # loop through the upcomming meetings
+        for post in futureMeetings:
+            meeting = Meeting(
+                title=self._parse_title(post, False),
+                description=self._parse_description(post, False),
+                classification=self._parse_classification(post),
+                start=self._parse_start(post, False, response),
+                end=self._parse_end(post, False, response),
+                all_day=self._parse_all_day(post, False),
+                location=self._parse_location(post, False, response),
+                links=self._parse_links(post, False),
+                source=self._parse_source(response)
+            )
+
+            meeting["status"] = self._get_status(meeting)
+            meeting["id"] = self._get_id(meeting)
+
+            yield meeting
+
+    """
+    To prevent rewriting the same functions for two cases (past/future),
+    we pass in an 'isPast' boolean to determine which case
+    """
+    def _parse_title(self, item, isPast):
+        """Parse or generate meeting title."""
+        if isPast:
+            title = item.xpath(".//h2[@class='entry-title']/a/text()").get()
+        else:
+            title = item.xpath("./a/text()").extract()[0]
+
+        return title
+
+    def _parse_description(self, item, isPast):
+        """Parse or generate meeting description."""
+        if isPast:
+            description = item.xpath("./div[@class='entry-content']/p/text()").get()
+
+            # some meeting minutes don't have descriptions
+            if not description:
+                description = ""
+        else:
+            # upcomming meetings don't have descriptions at all
+            description = ""
+
+        return description
+
+    def _parse_classification(self, item):
+        """Parse or generate classification from allowed options."""
+        # Northwest Home Equity Assurance Program is governing commission appointed by the mayor
+        return COMMISSION
+
+    def _parse_start(self, item, isPast, response):
```

Comment:
There's a lot going on in these methods, it might be cleaner to split out upcoming and past parsing into two functions, since it's essentially already doing that

Reply:
Good call we’ll do that.